openml-h2o/src/main/java/com/feedzai/openml/h2o/params/ParametersBuilderUtil.java

```diff
@@ -128,9 +141,68 @@ private ParametersBuilderUtil() { }
                 .filter(modelParam -> !"calibrate_model".equals(modelParam.getName()))
                 // We already pass a seed always whenever possible.
                 .filter(modelParam -> !"seed".equals(modelParam.getName()))
+                // We only want the fields that are present in the list of fields of the algorithm class.
+                .filter(modelParam -> usefulFields.contains(modelParam.getName()))
                 .collect(Collectors.toSet());
     }
 
+    /**
+     * Returns the fields that are set in the given {@linkplain ModelParametersSchemaV3 algorithmClass} as useful for
+     * the algorithm.
+     *
+     * @param algorithmClass The algorithm class to get the useful fields.
+     * @param allParameters  All the parameters to be added if useful fields is not provided by the class.
+     * @return The {@link List} with the useful fields for the given algorithm class.
+     * @since 1.0.7
+     */
+    private static List<String> getUsefulFields(final Class<? extends ModelParametersSchemaV3> algorithmClass,
+                                                final Set<String> allParameters) {
+
+        final Optional<Field> fieldsField = Arrays.stream(algorithmClass.getDeclaredFields())
+                .filter(field -> isStatic(field.getModifiers()))
+                .filter(field -> "fields".equalsIgnoreCase(field.getName()))
+                .findFirst();
```

Comment:
Nitpick: I find that using `Optional#isPresent` is a bit of an antipattern, as you're breaking the functional flow, usually without a need.

So, my suggestion would be to avoid calling `findFirst` at this point, then use a `map` to call `getStringArrayStaticFieldContent`, and then do `findFirst().map(<toSingletonList>).orElse(<emptyList>)`

Reply:
It's much nicer now. "Oneliner" :trophy: